### PR TITLE
Pipeline cleanup

### LINKED ISSUE #14: write a catch for uv sites with too much data
Some uv sites have so much data that the call to NWIS takes so long, but does not fail. For example, in the latest run, partition 240 ran overnight (tried two times) and still was not successful. Wonder if there's a way to time a call out after XX minutes or something? The problem is that the call to `whatNWISdata` does not give an accurate representation of what data are available for a given site. 

Here is partition 240, and site `405356112205601` has high frequency data at multiple depths. Most sites that cause problems are high frequency lake sites with multiple depths. This site is Great Salt Lake. I dropped that site number from the partition and completed the partition 240 download outside of `loop_tasks`, but a better solution should be implemented. The easiest solution would be to filter this site. 

image

### LINKED ISSUE #18: Follow new pull patterns in flow pipeline
While developing the [national-flow-observations](https://github.com/USGS-R/national-flow-observations) pipeline, Lindsay made some change to the way date is used in creating the pull task ID,  what functions are used to pull the data from NWIS, and what dates are use in the pulls from NWIS. I think these are safer, both in terms of unnecessary rebuilds, and are more fool-proof, since I myself have forgotten to change end date when I wanted to pull the most recent data. 

In the flow pipeline, these are the major changes I want to implement:
- trigger the rebuild by [manually changing the date](https://github.com/USGS-R/national-flow-observations/blob/master/10_nwis_pull.yml#L28) which gets appended to the pull ID. 
- use [readNWISdv and readNWISuv](https://github.com/USGS-R/national-flow-observations/blob/master/10_nwis_pull/src/nwis_pull.R#L78-L99) with end date set as "". readNWISdata does not allow for not specifying end date.

### LINKED ISSUE #19: Move data targets to getters.yml to avoid double builds
Similar to the solution implemented in[ delaware-model-prep](https://github.com/USGS-R/delaware-model-prep/blob/master/getters.yml), move all targets built with `get` statements to `getters.yml`. I started this process when adding new data in `4_other_sources`, but did not do for the whole pipeline.

### PR DESCRIPTION
**Pull request by limnoliver**
This PR cleans up several outstanding issues, including:

1. Including new EcoSHEDS file sent by Jeff Walker on 7/23 that updates through June 2020.
2. Moves all `gets` to `getters.yml` to avoid unnecessary builds per Alison's suggestion, and closes #19 
3. Follows new pull patterns developed in the national-flow-observations pipeline, which are mainly using `readNWISuv` and `readNWISdv` so we can use open start and end times to get all data (avoids having to reset dates, which I have forgotten in the past!). Also uses dummy dates to trigger repull, and uses those dates in pull IDs. This closes #18 
4. Limits NWIS pull to streams and springs, which avoids the largest lake/reservoir sites that were causing problems. This closes #14 - and I suggest adding an NWIS pull to [`lake-temperature-model-prep`](https://github.com/USGS-R/lake-temperature-model-prep) to add value to that dataset, but to also limit lake pulls to only areas where you need them. 
5. Repulls the data!

Of course, tinkering around caused new issues:

1. We are missing WQP sites that have no state label, because of a switch to having to loop through states to make a `whatWQPdata` call to retrieve observation counts to partition data pulls. Some of these _should_ be missing because they are outside of the U.S. and would get filtered downstream anyway. However, some of these are in the U.S. and truly just missing state labels. See #22 for a description and possible solutions.
2. UV data munge was causing memory failures in R. My solution was to reduce down to daily mean values in the combine step, so that the raw data is not preserved in the shared cache. I think this approach is ok, since we have a reproducible pipeline + are not using the raw data.

### CODE REVIEW

File: 5_data_munge/src/munge_wqp_files.R [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureValue`
+  ))
+  
+  dat <- mutate(dat, sample_depth_unit_code = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
```
Comment (aappling-usgs) [inline 5_data_munge/src/munge_wqp_files.R:38]: This guidance in `?case_when` makes me think you could get away with a simpler code block above.
```r
# Like an if statement, the arguments are evaluated in order, so you must
# proceed from the most specific to the most general. This won't work:
case_when(
  TRUE ~ as.character(x),
  x %%  5 == 0 ~ "fizz",
  x %%  7 == 0 ~ "buzz",
  x %% 35 == 0 ~ "fizz buzz"
)
```

So I'm fairly confident you could do
```r

dat <- mutate(dat, sample_depth_unit_code = case_when(
     !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
     !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
     !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
     TRUE ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
))
```

Comment (aappling-usgs) [reply 5_data_munge/src/munge_wqp_files.R:38]: double checking with a more similar example:
```r
> tibble(a=c(1, NA, NA, NA, 1), b=c(NA, 2, NA, 2, NA), c=c(NA, NA, 3, 3, 3)) %>%
+     mutate(best = case_when(!is.na(a) ~ a, !is.na(b) ~ b, !is.na(c) ~ c, TRUE ~ 0))
# A tibble: 5 x 4
      a     b     c  best
  <dbl> <dbl> <dbl> <dbl>
1     1    NA    NA     1
2    NA     2    NA     2
3    NA    NA     3     3
4    NA     2     3     2
5     1    NA     3     1
```

File: 1_wqp_pull/out/wqp_data_summary.csv [hunk 2]
```diff
@@ -1,2 +1,2 @@
-"n_obs","n_sites"
-51620919,355286
+n_obs,n_sites
+49640108,340901
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 1_wqp_pull/out/wqp_data_summary.csv:2]: Yes - this magnitude was expected based on my comparison with the old pull and missing state codes.

File: 1_wqp_pull/src/get_wqp_data.R [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    time_end <- Sys.time()
+    time_diff <- time_end - time_start
+    
+    return(list(time_diff, wqp_dat, 'readWQPdata'))
+  }
+    
+    wqp_out <- try(wqp_post_try(wqp_args))
+    
+    if (class(wqp_out) == 'try-error') {
+      message("Error with call to POST, trying dataRetrieval::readWQPdata")
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 1_wqp_pull/src/get_wqp_data.R:44]: This was fixing an old behavior (occasional fails of the POST call) that could probably be removed. It looked like all calls this time around were successful through POST (prints a message during the pull about which was used).

File: 2_nwis_pull/out/nwis_uv_summary.csv [hunk 4]
```diff
@@ -1,2 +1,2 @@
-"n_obs","n_sites"
-83725239,919
+n_obs,n_sites
+656498,617
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 2_nwis_pull/out/nwis_uv_summary.csv:2]: Yes, that's my only explanation for gain in uv inventory, but loss in data (sites get filtered after inventory).

File: 4_other_sources.yml [hunk 5]
```diff
@@ -10,20 +10,20 @@ sources:
...
   4_other_sources/out/ecosheds_data.rds.ind:  
     command: unzip_merge_values(
-      zip_ind = '4_other_sources/in/sheds-public-data-20200622.zip.ind',
+      zip_ind = '4_other_sources/in/sheds-public-data-20200723.zip.ind',
       out_ind = target_name)
       
   4_other_sources/out/ecosheds_sites.rds.ind:
     command: unzip_extract_sites(
-      zip_ind = '4_other_sources/in/sheds-public-data-20200622.zip.ind',
+      zip_ind = '4_other_sources/in/sheds-public-data-20200723.zip.ind',
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 4_other_sources.yml:26]: Yeah saw that mistake as well, did not mean to commit the first time around!

**Comment by wdwatkins** [comment]
@limnoliver I'm ready to run the pipeline with the new sites, if this PR is ready to go

**Pull request merged**